ip: Fix error when DHCP with auto IP address on STP enabled bridge

cathay4t opened the pull request:
When DHCP enabled with auto IP address on STP enabled bridge, nmstate
will fail with verification error:

    Verification failure: br0.interface.ipv4.address desire '[]',
    current 'null'

The root cause is STP suspended the DHCP action which cause current
state shows null IP address. And nmstate incorrectly treat [] != null
for IP address.

Fixed in `sanitize_current_for_verify()` to set empty array if None.

To reproduce this problem, we just enable DHCP with auto IP address
where no DHCP server exists. Integration test case included.

liangwen12year commented:
I did not spot any problem in this PR.

ffmancera commented on tests/integration/dynamic_ip_test.py:
```diff
@@ -1902,3 +1902,31 @@ def test_set_dhcp_host_name_and_remove(dhcpcli_up):
     assertlib.assert_state_match(desired_state)
+
+
+def test_auto_ip_with_pre_exist_address_without_dhcp_srv(eth1_up):
```
Just a nit: could we make this test as tier1? Thanks!

cathay4t replied:
marked. Thanks!

cathay4t committed:
ip: Fix error when DHCP with auto IP address on STP enabled bridge

When DHCP enabled with auto IP address on STP enabled bridge, nmstate
will fail with verification error:

    Verification failure: br0.interface.ipv4.address desire '[]',
    current 'null'

The root cause is STP suspended the DHCP action which cause current
state shows null IP address. And nmstate incorrectly treat [] != null
for IP address.

Fixed in `sanitize_current_for_verify()` to set empty array if None.

To reproduce this problem, we just enable DHCP with auto IP address
where no DHCP server exists. Integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>
